roles/variables/vars/main.yml

```diff
@@ -507,12 +507,16 @@ kafka_broker_properties:
       confluent.basic.auth.credentials.source: USER_INFO
       confluent.basic.auth.user.info: "{{schema_registry_ldap_user | default('sr') }}:{{schema_registry_ldap_password | default('pass')}}"
   sr_oauth:
-    enabled: "{{ kafka_broker_schema_validation_enabled and 'schema_registry' in groups and rbac_enabled and schema_registry_oauth_enabled }}"
+    enabled: "{{ kafka_broker_schema_validation_enabled and 'schema_registry' in groups and schema_registry_oauth_enabled }}"
     properties:
       confluent.bearer.auth.credentials.source: OAUTHBEARER
       confluent.bearer.auth.issuer.endpoint.url: "{{ oauth_token_uri }}"
       confluent.bearer.auth.client.id: "{{ schema_registry_oauth_user }}"
       confluent.bearer.auth.client.secret: "{{ schema_registry_oauth_password }}"
+  sr_oauth_scope:
```

Comment:
Little doubt in the lines 514 and 515. Why are we using sr oauth user here, these configs are for broker talking to sr, so shouldn't broker's user i.e the oauth_client_id be used here

Reply:
broker does not have the permissions to talk to sr, it uses the sr client to validate, similar to how we use sr_ldap_user in the above condition. For SR to talk to broker, we provide it with the rolebidings

Reply:
agreed this is similar to how we did it in ldap. but still I find this weird behaviour that broker doesnt have permissions to talk to sr and hence it uses sr user itself to talk to sr

Reply:
actually it does have permissions since it is the super user, now I remeber we use this because we don't want to use the super user as client